### PR TITLE
PIMS-1373 ParcelBuildings Restructure Entities

### PR DESCRIPTION
**Pull request by dbarkowsky**
## 🎯 Summary

  
[PIMS-1371](https://citz-imb.atlassian.net/browse/PIMS-1373?atlOrigin=eyJpIjoiZTY0OGU1Y2MyYjdmNDk5MWIzYTA5MjkwYmRkYTRkMzUiLCJwIjoiaiJ9)

## Changes
- Removed AccessRequest entity
- Changed UpdatedOn columns to be automatically updated
- Remove ParcelBuilding entity
- Move PID/PIN fields to Property entity so both parcels and buildings will have them
- Create a view (BuildingRelations) that can be used to reference property/building associations
- Add new migration files to accommodate changes

## Testing
1. Run migrations (can just start API)
2. Check database to see the new changes.

## Future Work
- Update ETL so that the buildings are populated with PID/PIN if they have associated parcels in old PIMS

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/src/typeorm/Migrations/1710191053571-AddBuildingRelationsView.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  name = 'AddBuildingRelationsView1710191053571';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`CREATE MATERIALIZED VIEW "building_relations" AS 
+    SELECT b.id AS "building_id", b.pid, b.pin, p.id as "parcel_id"
+    FROM building b
+    LEFT OUTER JOIN parcel p
+    ON b.pid = p.pid OR b.pin = p.pin;
+  `);
+    await queryRunner.query(
```
Comment (GrahamS-Quartech) [inline express-api/src/typeorm/Migrations/1710191053571-AddBuildingRelationsView.ts:13]: Is writing directly to this table correct? Seems like something that should be auto-generated.

File: express-api/src/typeorm/Entities/views/PropertyView.ts
Comment (GrahamS-Quartech) [inline express-api/src/typeorm/Entities/views/PropertyView.ts:1]: This makes sense 👍

File: express-api/src/typeorm/Migrations/1710191053571-AddBuildingRelationsView.ts (hunk 1 above)
Comment (dbarkowsky) [reply express-api/src/typeorm/Migrations/1710191053571-AddBuildingRelationsView.ts:13]: This was what the TypeORM package generated. I didn't manually write this.

**Comment by dbarkowsky** [comment]
Made a few changes from [PIMS-1375](https://citz-imb.atlassian.net/browse/PIMS-1375?atlOrigin=eyJpIjoiMzJjOGNlMWFjYTM2NGU0N2I3OWQyMmFiMjQ0YmUyZDkiLCJwIjoiaiJ9).

There weren't any notable updates to the Parcel service, but I tried to clean out some old interfaces and add references to Buildings and Parcels to more easily access their Evaluations and Fiscals.

**Review by TaylorFries — APPROVED** [review]
migrations work as expected and database updates look good.

**Pull request merged**